### ISSUE TITLE
Add iNaturalist dataset

### ISSUE THREAD

**Issue by vfdev-5** [pytorch/vision#3292]
## 🚀 Feature

A feature request to discuss about about adding iNaturalist dataset wrapper of ImageFolder (maybe, even without downloading tools). 

## Motivation

We can see in the research code custom dataset implementation which can be also in torchvision. For example, here:

https://github.com/facebookresearch/deit/blob/ab5715372db8c6cad5740714b2216d55aeae052e/datasets.py#L13

Being also aware of our not-yet-defined procedure on which datasets to add, feel free to close the FR if we can not move forward. 

cc @pmeier

**Comment by pmeier** [comment]
It seems to be cited [159 times](https://scholar.google.com/scholar?cites=8277973545413149328). Given that this is not a general purpose classification dataset, I think it is reasonable popular. @vfdev-5 do you have insights if this is still used in contemporary publications or i seven picking up in popularity?

**Comment by vfdev-5** [comment]
According to scholar, there are already 9 papers citing the dataset in 2021. Code cited above is for Deit (12/2020) paper.
Looks like a popular dataset. Let's see other opinions here before going further.

**Comment by pmeier** [comment]
Just to be clear: I'm in favor of adding this.

**Comment by arch-user-france1** [comment]
There is a mistake in the [documentation](https://pytorch.org/vision/stable/generated/torchvision.datasets.INaturalist.html#torchvision.datasets.INaturalist).

The module is spelled lowercase (`torchvision.datasets.inaturalist`). There is no `torchvision.datasets.INaturalist`.

**Comment by pmeier** [comment]
@arch-user-france1 There is no mistake in the documentation since it doesn't describe the module, but rather the class:

https://github.com/pytorch/vision/blob/120e7af6466190b754cf3026c685a5d31561da90/torchvision/datasets/inaturalist.py#L31

Seeing the `AttributeError`'s above, I guess something is wrong with your environment. Please open a new issue following the steps in the form so we can have a look.

**Comment by arch-user-france1** [comment]
> @arch-user-france1 There is no mistake in the documentation since it doesn't describe the module, but rather the class:
> 
> https://github.com/pytorch/vision/blob/120e7af6466190b754cf3026c685a5d31561da90/torchvision/datasets/inaturalist.py#L31
> 
> Seeing the `AttributeError`'s above, I guess something is wrong with your environment. Please open a new issue following the steps in the form so we can have a look.

Ah, forget it. I had set up a buggy environment. Fixed it... Weird, though.

**Comment by arch-user-france1** [comment]
There is a memory leak in the dataset or DataLoader - is it supposed to use more than 30GB of RAM? It does not free any memory, so a dataset of a this size would be unusable. Nobody has 200GB RAM...
I have not spotted any issue in my code, and I have seen a stackoverflow question describing the same problem.

**Comment by nhewadehigah** [comment]
Hi Do anyone know how to access the supercategory label in 2021 version. In tensorflow it has supercategory label but not seen in pytorch.
